### PR TITLE
Doc updates for promoting csi plugins on windows to stable

### PR DESCRIPTION
**Pull request by marosset**
PR for promoting CSI plugins to stable for windows.

Related to https://github.com/kubernetes/enhancements/issues/1122

/sig windows

### CODE REVIEW

**Comment by sftim** [comment]
This'll need a tech review from SIG Windows. All the changes look reasonable but I haven't a handy way to check them.

**Comment by marosset** [comment]
/assign @ddebroy @jsturtevant @jingxu97 

Can one of you help with the SIG windows review?

File: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md [hunk 1]
```diff
@@ -247,21 +247,14 @@ DaemonSets and StatefulSets. CSI plugins handle a wide range of volume
...
-CSI node plugins is supported using
-[csi-proxy](https://github.com/kubernetes-csi/csi-proxy), a community-managed,
-stand-alone binary that needs to be pre-installed on each Windows node. Please
-refer to the deployment guide of the CSI plugin you wish to deploy for further
-details.
+backup/restore of persistent data using snapshots and cloning.
+
+CSI plugins communicate with a CSI node plugin which performs the local storage operations.
+On Windows nodes the community-managed [csi-proxy](https://github.com/kubernetes-csi/csi-proxy)
+is used.
```
Comment (ddebroy) [inline content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:254]: how about: "is used" => "is typically used by CSI node plugins."

File: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md [hunk 2]
```diff
@@ -247,21 +247,14 @@ DaemonSets and StatefulSets. CSI plugins handle a wide range of volume
...
+
+Please refer to the following [installation steps](https://github.com/kubernetes-csi/csi-proxy#installation)
```
Comment (ddebroy) [inline content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:256]: I would prefer to continue to have users refer to their environment/CSI plugin docs first as the environment may be already installing csi proxy for them. So how about: "Please
refer to the deployment guide of the CSI plugin you wish to deploy for further details around installation. You may also refer to the following ..."

**Review by ddebroy — COMMENTED** [review]
Thanks for the PR! A couple of suggestions.

**Comment by ashnehete** [comment]
Hello @ddebroy  :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday 20th of July, 11:59 PM PDT. Thank you!

**Comment by ddebroy** [comment]
I think overall this PR is ready. Just a couple of suggestions above for @marosset.

**Comment by marosset** [comment]
I'll make the suggested updates today.

File: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md (hunk 1 above)
Comment (marosset) [reply content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:254]: How does this sound:

CSI plugins communicate with a CSI node plugin which performs the local storage operations.
On Windows nodes CSI node plugins typically call APIs exposed by the community-managed
[csi-proxy](https://github.com/kubernetes-csi/csi-proxy) which handles the local storage operations.

This adds a few more details for folks who don't click through but also captures your updates (I hope)

File: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md (hunk 2 above)
Comment (marosset) [reply content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md:256]: Updated

**Comment by marosset** [comment]
Also note that there is not a feature-gate for this feature (https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1122-windows-csi-support) so I'm not updating feature-gates.md here.

**Comment by sftim** [comment]
/lgtm
for SIG Docs

**Pull request merged**